sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIOTranslation.java

```diff
@@ -274,10 +274,16 @@ public TypedRead<?> fromConfigRow(Row configRow, PipelineOptions options) {
         if (queryTempDataset != null) {
           builder = builder.setQueryTempDataset(queryTempDataset);
         }
-        String queryTempProject = configRow.getString("query_temp_project");
-        if (queryTempProject != null) {
-          builder = builder.setQueryTempProject(queryTempProject);
+
+        if (TransformUpgrader.compareVersions(updateCompatibilityBeamVersion, "2.57.0") >= 0) {
```

Comment:
This flag is for people to set the version back into the past. I don't think it works for this use case of making sure it is far enough into the future.

Reply:
The problem case will be when this flag is not set at all, which will be the default case. So can we make things work by default when the flag is not set?

Reply:
flag should be set by TransformUpgrader here https://github.com/apache/beam/blob/1037edec0dd3fee83556f3ff25c2d0fab752a3d1/sdks/java/core/src/main/java/org/apache/beam/sdk/util/construction/TransformUpgrader.java#L251-L259

Reply:
Yeah, this flag is set by the client when upgrading via the TransformUpgrader (client) so BQ transform just have to check it when upgrading and skip new fields (and set defaults).